src/sender_queue/mod.rs

```diff
@@ -284,13 +346,19 @@ where
         self
     }
 
+    pub fn last_epochs(mut self, last_epochs: BTreeMap<D::NodeId, D::Epoch>) -> Self {
+        self.last_epochs = last_epochs;
+        self
+    }
```

Comment:
Do we really need that in the builder? A joining node should receive all the required information in one of the following batches, shouldn't it? (Also unsure about `peer_epochs` and `outgoing_queue`.)

Reply:
To be honest, I'm not sure we need the builder at all. But if we keep it in case the user wishes to replace the managed algorithm or change peer epochs (questionable), we could leave only `peer_epochs` out of the three - `outgoing_queue`, `peer_epochs` and `last_epochs` in the builder.